ui/lib/apps/TopSQL/pages/List/List.tsx

```diff
@@ -107,40 +107,44 @@ export function TopSQLList() {
     instance,
     timeRange,
     timeWindowSize,
-    topN,
     autoRefreshSeconds,
     containerRef,
     containerWidth,
+    updateTopSQLData,
   ])
 
   // fetch data when instance id / time window size update
   useEffect(() => {
     handleUpdateTopSQLData()
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Do we have some safer way to do this magic? I mean, something that will not be reported by the lint while simple enough. I'm not an expert of React hook. How does other React hook users deal with this kind of case? @shhdgit

Reply:
OK, this is not a good scenario for the use of side effects. According to this article: https://stackoverflow.com/questions/58866796/understanding-the-react-hooks-exhaustive-deps-lint-rule

In the similar case, I should call the function directly when the event is triggered. Oh my react.